clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -91,12 +91,31 @@ public int throttleTimeMs() {
     public Map<String, Errors> errors() {
         Map<String, Errors> errors = new HashMap<>();
         for (MetadataResponseTopic metadata : data.topics()) {
+            if (metadata.name() == null) {
+                throw new NullPointerException("Use errorsByTopicId() when manage topic using topic id");
```

Comment:
We should never get here through the public APIs right? We could use IllegalStateException.